### PR TITLE
Create wrapper functions

### PR DESCRIPTION
**Pull request by emmakatherina**
Fixes issue #70 

Please review and/or merge.

### CODE REVIEW

File: gap/main.gi [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    end
+);
+
+InstallGlobalFunction( EchelonMatTransformationBlockwise,
+    function ( mat )
+        local result;
+        result := DoEchelonMatTransformationBlockwise( mat );
+        return rec(
+            vectors := result.vectors,
+            heads := result.pivotrows, #TODO ersetze mit heads
```
Comment (emmakatherina) [inline gap/main.gi]: This is in the other PR.

**Comment by emmakatherina** [comment]
I am not sure whether the commit history is correct (the commit 690ea6c is originally from another PR but after the new commits). But maybe this is just a different order and we don't get commits twice, now?

**Comment by ssiccha** [comment]
I guess the extra commit will disappear once we rebase this on top of master. Weird that it only got added "after" the README change.

**Comment by emmakatherina** [comment]
Is there anything speaking against merging the code now? Did I miss something?

**Comment by ssiccha** [comment]
No, you didn't miss something. I just didn't have the time yet to look through the whole PR. Sorry.

File: gap/main.gi (hunk 1 above)
Comment (ssiccha) [reply gap/main.gi]: The other PR was merged so please update this. Or am I missing something?

**Comment by ssiccha** [comment]
IMO the commit title `Create global functions` should be changed to `Create wrapper functions`. `global function` is a very general description and could mean anything. You can change this by doing `git rebase -i master` and choose the reword option.

Another minor point: you amended the requested changes into the last commit. It would have been nicer if you put them into the first commit. You can merge a commit into an arbitrary other commit by doing `git rebase -i master`. What I usually do is:
- create commits with the requested changes and call them
  - `SQUASH: ...`
  - `SQUASH: ...`
  - ...
- Then call `git rebase -i master` which opens an editor. There you move each new commit under the respective commit that you want to merge it into. If you would quit now this would reorder the commits. Don't quit yet though. Select the `squash` instead of the `pick` option for the new commits. This lets you merge them into the commit directly above them.

File: gap/main.gd [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Declare global functions of this package
+
+DeclareGlobalFunction( "DoEchelonMatTransformationBlockwise", "Calculates echelon form and all other informations" );
```
Comment (ssiccha) [inline gap/main.gd]:
```suggestion
DeclareGlobalFunction( "DoEchelonMatTransformationBlockwise", "Calculates echelon form and all other information" );
```
"Information" has no plural form in English.

File: gap/main.gi [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+        dim := DimensionsMat( mat );
+        if Size(options) < 4 then
+            numberChopsHeight := GAUSS_calculateChops( dim[1] );
+            numberChopsWidth := GAUSS_calculateChops( dim[2] );
+        else
+            numberChopsHeight := options[3];
+            numberChopsWidth := options[4];
+        fi;
+        Info(InfoGauss, 1, "The matrix has a height of ", dim[1],
```
Comment (ssiccha) [inline gap/main.gi]: IMO we should only print the number of chops in both directions. The user already knows the size of the matrix. Unnecessary extra information tends to make the important information less visible.

Also please add an extra info statement that warns the user if any of `numberChops..` is equal to 1. In that case I think we will get almost no speed gain by using multiple threads.

File: gap/main.gi (hunk 1 above)
Comment (ssiccha) [reply gap/main.gi]:
```suggestion
            heads := result.heads,
```

**Comment by emmakatherina** [comment]
Thank you very much for your detailed explanation of the rebase command! I just learned a lot of usefull stuff. Unfortunately git does complain about merge conflicts, so I added the new changes to the last commit anyway...

The commit message is also changed.

**Comment by ssiccha** [comment]
Yeah, I wouldn't try to resolve all these merge conflicts then. That way lies madness! :)

**Pull request merged**